projects/core/src/cms/page/base-page-meta.resolver.ts

```diff
@@ -48,21 +60,37 @@ export class BasePageMetaResolver
    */
   protected breadcrumb$: Observable<BreadcrumbMeta[]> = combineLatest([
     this.homeBreadcrumb$,
-    defer(() => this.routingPageMetaResolver.resolveBreadcrumbs()),
+    defer(() => this.routingPageMetaResolver?.resolveBreadcrumbs()),
   ]).pipe(
     map((breadcrumbs) => breadcrumbs.flat()),
     shareReplay({ bufferSize: 1, refCount: true })
   );
 
-  resolveTitle(): Observable<string> {
+  resolveTitle(): Observable<string | undefined> {
     return this.title$;
   }
 
-  resolveBreadcrumbs(): Observable<BreadcrumbMeta[]> {
+  resolveBreadcrumbs(): Observable<BreadcrumbMeta[] | undefined> {
     return this.breadcrumb$;
   }
 
   resolveRobots(): Observable<PageRobotsMeta[]> {
     return this.robots$;
   }
+
+  resolveCanonicalUrl(
+    url?: string,
```

Comment:
[medium] Since this observable already depends on the Router, IMO we should can drop the `url` param here. The `BasePageMetaResolver` by design uses the _current URL_. And only `PageLinkService` should be used with specific URLs (i.e. product variant url).

Moreover, for clarity of the API, I would:
- make `url` a required param of `PageLinkService.getCanonicalUrl(url, options?)`
- use `this.winRef.location.href` explicitly in this method, but not in `PageLinkService`. For example:
   ```ts
   const currentUrl = this.winRef.location.href;
   this.pageLinkFactory!.getCanonicalUrl(currentUrl, options)
   ```

Reply:
Agreed on dropping the url from the `BasePageMetaResolver.resolveCanonicalUrl` arguments.
I do like to keep the winRef inside the pageLinkService so that the page resolver doesn't need to be concerened with this.